lib/core/util/mapper/mapper.dart

```diff
@@ -0,0 +1,28 @@
+import '../../../data/model/genre_model.dart';
+import '../../../data/model/movie_model.dart';
+import '../../../domain/entity/genre.dart';
+import '../../../domain/entity/movie.dart';
+
+class MapperClass {
```

Comment:
Why you are using this mapping class?

Reply:
this class is used to map a model from data layer to an entity in order to follow clean architecture guidelines, and make sure that domain layers and afterward presentation layers receive entities and not models.

Reply:
But you don't need to do it, you can define the entity con domain layer and on data layer you can create the model that extends from the domain class, like this:
domain:
    - movie_entity
data:
    - movie_model
    
 movie_entity
On movie entity you define the structure that you want to have or use
```
class MovieEntity {
     final int id;
     final String name;

     MovieEntity({
          required this.id,
          required this.name,
    });
}
```
movie_model
On movie model you extends the structure that is defined on movie_entity
```
class MovieModel extends MovieEntity {
    MovieModel({
          required int id,
          required String name,
    }): super(
          id: id,
          name: name,
    );
    
    factory MovieModel.fromJson(...) => ...
}
```

Reply:
done! really useful advice fran! thanks!